modules/lfs/utils.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package lfs
+
+import (
+	"io"
+	"strconv"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// ReadLFSPointerFile will return a partially filled LFSMetaObject if the provided reader is a pointer file
+func ReadLFSPointerFile(reader io.Reader) (*models.LFSMetaObject, *[]byte) {
+	buf := make([]byte, 1024)
+	n, _ := reader.Read(buf)
+	buf = buf[:n]
+
+	isTextFile := base.IsTextFile(buf)
+	if isTextFile && setting.LFS.StartServer {
+		return IsLFSPointerFile(&buf), &buf
+	}
+
+	return nil, &buf
+}
+
+// IsLFSPointerFile will return a partially filled LFSMetaObject if the provided byte slice is a pointer file
+func IsLFSPointerFile(buf *[]byte) *models.LFSMetaObject {
+	if setting.LFS.StartServer {
```

Comment:
Please change this function to make it better readable with reverse conditions, ex:
```go
if !setting.LFS.StartServer {
     return nil
}
headString := string(*buf)
if !strings.HasPrefix(headString, models.LFSMetaFileIdentifier) {
    return nil
}
// etc
```